### DISCUSSION TITLE
DAM hourly LMPs API only returns last 3 months

### DISCUSSION THREAD

**Discussion by tamird** [ercot/api-specs#24] [General]
Hello. I'm trying to fetch [DAM hourly LMPs](https://apiexplorer.ercot.com/api-details#api=pubapi-apim-api&operation=getData_30) but I can't seem to get anything before ~3.5 months ago. At the same time, much older data is available on the [public archive](https://data.ercot.com/data-product-archive/NP4-183-CD).

Would it be possible to make the entire archive available via the API?

**Comment by mackermann-ercot** [comment]
Hello Tamir, 

We're glad to see you have joined the discussions!  Our primary objective for the Public API  project is to provide historical archives for our public EMIL products that are no longer available through our main ercot.com site.  The file archive provides EMIL products for the last seven years (and in some cases much longer), however we only provide API access to the data within those products from mid December of 2023 year (during the Beta release).

However, If there is enough interest to provide additional historical data for selected reports, we may consider loading data prior to the beta release.

Regards,
mackermann, ERCOT

**Reply by ianzhg** [reply]
Tamir,

This is a very useful tool. Love to see you guys developing it.

I was also trying to access data for RT pricing for each substation, and RT pricing for resource nodes, load zones, and hubs, but the data only goes from mid-Decemebr.

Would love to see data loaded for previous time because it would be extremely for us to understand the RT pricing risks given all the turbulent pricing we have had in the last couple of years.

Cheers,
Ian

**Reply by mackermann-ercot** [reply]
Hi Ian, 

Thanks for the encouraging words!  I will pass along your request for an extended data selection for the _LMPs by Resource Nodes, Load Zones and Trading Hubs_ product to give it another up-vote for future enhancements.

Regards, 
mackermann, ERCOT

**Reply by oorlandini** [reply]
Hi ya'll, I just started working with this awesome API and between @mackermann-ercot and @CaffieneLab have an extremely convenient little python script going to fill some holes and keep our database updated. Thank you so much for your work on this! 

I'd also like to second how deeply I would adore making older records available via this great API. I don't know who I should send flowers and/or donuts to in Taylor to help persuade, but this tool is outstanding and we would definitely use it proportionate to any further effort put into it.

**Comment by manu1151** [comment]
Hello mackermann,

This would be extremely useful, I've had the same issue myself and would benefit enormously from having access to prior data (< Dec 2023).

Thank you!

Manuel

**Comment by dpyrek-ea** [comment]
Hello Mackermann,

I wanted to add a comment supporting extended data selection. I ran into similar issue pulling data from the 60 Day SCED Gen Resource Data API back through 2023. The API has already been an incredibly useful tool, and would be augmented greatly with additional data to avoid scraping csvs from the historical archives.

Thank you very much!
Daniel

**Comment by DMJ-CV-91913** [comment]
I would like to ask ERCOT to extend the API historical datasets or corpus back to Nov 1, 2021  to allow for a robust experience and provide useful data and efficiencies to the developer community supporting market participants either for modeling, researching, reporting, risk assessment, planning, etc.   This date is chosen as prior data other than load data, doesnt have any relevant statistical significance given the dynamic changes of the grid( powerflows, intermittency, constraints, ramp rates).  The date provides almost 3 years of data(>post covid), which is enough for almost any market participant needs.

**Comment by haqureshi** [comment]
How can i call Find Transaction Method of Retail API 
(https://misapi.ercot.com/retailAPI/)

Please help

**Comment by will-strauch-toce** [comment]
Adding a comment to support historical data additions!
